### PR TITLE
Add mentionable teams to tributeValues and change team mention rules to gh's style

### PR DESCRIPTION
**Pull request by a1012112796**
As title, example view:
image

TODO:

- [x] Maybe need change the rule for team mention to @{{OrgName}}/{{TeamName}}  to protect a team have same name with a user.
- [x] have no idea about how to fix avatar bug now (use org's avator now)

### CODE REVIEW

**Comment by 6543** [comment]
@a1012112796 avatar bug: use the same icon as with pullreview-requested-teams:

Screenshot at 2020-10-18 21-43-47
?

File: templates/base/head.tmpl [hunk 1]
```diff
@@ -45,7 +45,7 @@
 				EventSourceUpdateTime: {{NotificationSettings.EventSourceUpdateTime}},
 			},
 			PageIsProjects: {{if .PageIsProjects }}true{{else}}false{{end}},
-      {{if .RequireTribute}}
+      	{{if .RequireTribute}}
```
Comment (silverwind) [inline templates/base/head.tmpl:48]:
```suggestion
      {{if .RequireTribute}}
```

Comment (silverwind) [inline templates/base/head.tmpl]:
```suggestion
			{{if .RequireTribute}}
```

File: templates/base/head.tmpl [hunk 2]
```diff
@@ -55,8 +55,11 @@
 				['{{.Name}}', {key: '{{.Name}} {{.FullName}}', value: '{{.Name}}',
 				name: '{{.Name}}', fullname: '{{.FullName}}', avatar: '{{.RelAvatarLink}}'}],
 				{{ end }}
+				{{ range .MentionableTeams }}
+					['{{.Name}}', {key: '{{.Name}}', value: '{{.Name}}', name: '{{.Name}}'}],
+				{{ end }}
 			]).values()),
-			{{end}}
+		{{end}}
```
Comment (silverwind) [inline templates/base/head.tmpl]:
```suggestion
			{{end}}
```

**Comment by silverwind** [comment]
> use the same icon as with pullreview-requested-teams

I'd use the org icon if possible

**Comment by lafriks** [comment]
Imho we should prefix them with org name and other possibility is to use specific group icon

**Comment by lunny** [comment]
I think maybe we should support mention anyone at first.

**Comment by a1012112796** [comment]
> I think maybe we should support mention anyone at first.

Do you means "@all" ?

**Comment by lunny** [comment]
> > I think maybe we should support mention anyone at first.
> 
> Do you means "@ALL" ?

No. I mean currently we can only mention the orgnizaion members. We should allow to mention anyone in this Gitea instance.

**Comment by a1012112796** [comment]
> > > I think maybe we should support mention anyone at first.
> > 
> > 
> > Do you means "@ALL" ?
> 
> No. I mean currently we can only mention the orgnizaion members. We should allow to mention anyone in this Gitea instance.
you can mention anyone now. but not support hint on ui  now. gitea only add all assignees and participants to the list now.

File: modules/markup/html.go [hunk 3]
```diff
@@ -568,7 +568,7 @@ func replaceContentList(node *html.Node, i, j int, newNodes []*html.Node) {
 	}
 }
 
-func mentionProcessor(ctx *postProcessCtx, node *html.Node) {
+func mentionProcessor(ctx *postProcessCtx, node *html.Node) { // TODO
```
Comment (a1012112796) [inline modules/markup/html.go]:
```suggestion
func mentionProcessor(ctx *postProcessCtx, node *html.Node) {
```

**Comment by silverwind** [comment]
I generally think those mentions need to be rewritten to use a AJAX call to obtain the mentionees. I think tribute.js already supports async loading of data. Doesn't need to be done in this PR but would be nice to have.

File: models/issue.go [hunk 4]
```diff
@@ -1838,30 +1838,43 @@ func (issue *Issue) ResolveMentionsByVisibility(ctx DBContext, doer *User, menti
...
+	resolved := make(map[string]bool, 10)
+	var mentionTeams []string
+
+	if err := issue.Repo.getOwner(ctx.e); err != nil {
+		return nil, err
+	}
+
+	repoOwnerIsOrg := issue.Repo.Owner.IsOrganization()
+	if repoOwnerIsOrg {
+		mentionTeams = make([]string, 5)
```
Comment (lafriks) [inline models/issue.go]:
```suggestion
		mentionTeams = make([]string, 0, 5)
```

**Review by lafriks — CHANGES_REQUESTED** [review]

**Review by lafriks — APPROVED** [review]

File: models/issue.go [hunk 5]
```diff
@@ -1892,10 +1905,10 @@ func (issue *Issue) ResolveMentionsByVisibility(ctx DBContext, doer *User, menti
 				}
 				if has {
 					checked = append(checked, team.ID)
-					resolved[team.LowerName] = true
+					resolved[issue.Repo.Owner.LowerName+"/"+team.LowerName] = true
 				}
 			}
-			if len(checked) != 0 {
+			if len(checked) > 0 {
```
Comment (6543) [inline models/issue.go]: reason to change this?

File: modules/references/references.go [hunk 6]
```diff
@@ -26,8 +26,8 @@ var (
 	// While fast, this is also incorrect and lead to false positives.
 	// TODO: fix invalid linking issue
 
-	// mentionPattern matches all mentions in the form of "@user"
-	mentionPattern = regexp.MustCompile(`(?:\s|^|\(|\[)(@[0-9a-zA-Z-_]+|@[0-9a-zA-Z-_][0-9a-zA-Z-_.]+[0-9a-zA-Z-_])(?:\s|[:,;.?!]\s|[:,;.?!]?$|\)|\])`)
+	// mentionPattern matches all mentions in the form of "@user" or "@org/team"
+	mentionPattern = regexp.MustCompile(`(?:\s|^|\(|\[)(@[0-9a-zA-Z-_]+|@[0-9a-zA-Z-_]+\/?[0-9a-zA-Z-_]+|@[0-9a-zA-Z-_][0-9a-zA-Z-_.]+\/?[0-9a-zA-Z-_.]+[0-9a-zA-Z-_])(?:\s|[:,;.?!]\s|[:,;.?!]?$|\)|\])`)
```
Comment (6543) [inline modules/references/references.go:30]: we should add tests for this regexp ...!

File: models/issue.go (hunk 5 above)
Comment (a1012112796) [reply models/issue.go]: Hmm, It's not usefull.
```suggestion
			if len(checked) != 0 {
```

**Review by 6543 — APPROVED** [review]

**Pull request merged**